### DISCUSSION TITLE
Get query result

### DISCUSSION THREAD

**Discussion by DenisKoshelev** [getredash/redash#6263] [Q&A] [answered]
Hello, i'm trying to get query result(/api/queries/<id>/results), and i get 2 answers from API:
1) return "400 bad request" for query with parameters
2) 200 for query without parameters, but i don't see job_id in the answer

url which i use look like that "https://myredash.ru/api/queries/1848/results?api_key=apiKey"

If i understand correctly for get query result - I need in first get job id from /api/queries/<id>/results, is it true?

Please help me to find my mistake, why do i get 400? Maybe error in my body or i need send something else?

```
async getData() {
  let headers = {'content-type': 'application/json'};
  let body = {'parameters' : {'p_direction' : ['Дизайн', 'Игры', 'Управление', 'Программирование', 'Маркетинг'], 'p_reasons' : 'По оплаченному счету'}};
  try {
      await fetch(this.getUrl(queryName), {method: 'POST', headers: headers, body: body})
          .then(async (response) => {
              return response.json();
          })
  } catch(e) {
      console.log(e);
  }
}
```

**Comment by gaecoli** [comment]
For `/api` you have to set `Authorization`.

**Reply by DenisKoshelev** [reply]
Hello, gaecoli, thank you for your answer.
I try to add api key in headers like this - 'Authorization': 'Key apiKey'
And i try add query api key and my profile api key.
But it is not work for me, i also get 400 answer.

Maybe you have different ideas?

**Comment by DenisKoshelev** [comment, marked as answer]
I was able get data.
This help me https://gist.github.com/arikfr/ab39499738976a61855f98cc62d39e52
My code:
```
const instance = axios.create({
    baseURL: `https://myredash.ru/`,
    headers: {
        'content-type': 'application/json; charset=UTF-8',
        Authorization: `Key apiKey`
    }
});

const url = `/api/queries/resultId/results`;
let response = await instance.post(url, options);
```
